File: src/main/scala/VectorClass.scala
//sealed trait Vector {
//  def keyRepresentation: String
//  def valueRepresentation: String
//}

case class PolarVector(length: Double, angle: Double) extends Vector {
  def keyRepresentation: String = s"Polar: Length=$length, Angle=$angle"
  def valueRepresentation: String = s"($length, $angle)"
}

case class CartesianVector(x: Double, y: Double) extends Vector {
  def keyRepresentation: String = s"Cartesian: X=$x, Y=$y"
  def valueRepresentation: String = s"($x, $y)"
}

class PolarVectorType extends VectorType {
  def create: Vector = PolarVector(0, 0)
  def clone(vector: Vector): Vector = vector.asInstanceOf[PolarVector].copy()
  def parse(s: String): PolarVector = {
    val parts = s.split(",")
    PolarVector(parts(0).toDouble, parts(1).toDouble)
  }
  def typeName: String = "Polar Vector"
}

class CartesianVectorType extends VectorType {
  def create: Vector = CartesianVector(0, 0)
  def clone(vector: Vector): Vector = vector.asInstanceOf[CartesianVector].copy()
  def parse(s: String): CartesianVector = {
    val parts = s.split(",")
    CartesianVector(parts(0).toDouble, parts(1).toDouble)
  }
  def typeName: String = "Cartesian Vector"
}

File: src/main/kotlin/VectorClass.kt
data class PolarVector(val length: Double, val angle: Double) : Vector {
    override val keyRepresentation: String get() = "Polar: Length=$length, Angle=$angle"
    val valueRepresentation: String get() = "($length, $angle)"
}

data class CartesianVector(val x: Double, val y: Double) : Vector {
    override val keyRepresentation: String get() = "Cartesian: X=$x, Y=$y"
    val valueRepresentation: String get() = "($x, $y)"
}

class PolarVectorType : VectorType {
    override fun create(): Vector = PolarVector(0.0, 0.0)
    override fun clone(obj: Vector): Vector = (obj as PolarVector).copy()
    override fun parse(input: String): PolarVector {
        val parts = input.split(",")
        return PolarVector(parts[0].toDouble(), parts[1].toDouble())
    }
    override val typeName: String = "Polar Vector"
}

class CartesianVectorType : VectorType {
    override fun create(): Vector = CartesianVector(0.0, 0.0)
    override fun clone(obj: Vector): Vector = (obj as CartesianVector).copy()
    override fun parse(input: String): CartesianVector {
        val parts = input.split(",")
        return CartesianVector(parts[0].toDouble(), parts[1].toDouble())
    }
    override val typeName: String = "Cartesian Vector"
}
